pkg/provider/scale_object.go

```diff
@@ -0,0 +1,35 @@
+package provider
```

Comment:
This probably needs a `scale_object_test.go` with corresponding unit tests for coverage.